web-ui/src/components/feedback_request_confirmation/FeedbackRequestConfirmation.jsx

```diff
@@ -1,28 +1,65 @@
-import React from "react";
+import React, { useContext } from "react";
 import Typography from "@material-ui/core/Typography";
 import CheckCircleIcon from '@material-ui/icons/CheckCircle';
-import {makeStyles} from '@material-ui/core/styles';
-
-
+import { makeStyles } from '@material-ui/core/styles';
+import { selectProfile } from "../../context/selectors";
+import { AppContext } from "../../context/AppContext";
+import { useLocation } from 'react-router-dom';
+import queryString from 'query-string';
+import DateFnsUtils from "@date-io/date-fns";
 import "./FeedbackRequestConfirmation.css";
-import {green} from "@material-ui/core/colors";
+import { green } from "@material-ui/core/colors";
+
+const dateUtils = new DateFnsUtils();
 
-const formatNamesAsList = (names) => {
-  return names.join(", ")
-}
 const useStyles = makeStyles({
-announcement: {
-  textAlign: "center"
-},
+  announcement: {
+    textAlign: "center"
+  },
 });
 
+let today = new Date();
+today = dateUtils.format(today, "yyyy-MM-dd");
+
 const FeedbackRequestConfirmation = (props) => {
   const classes = useStyles();
+  const { state } = useContext(AppContext);
+  const location = useLocation();
+  const query = queryString.parse(location?.search);
+  const forQuery = query.for?.toString();
+  const fromQuery = query.from?.toString();
+  const sendQuery = query.send?.toString();
+  const requestee = selectProfile(state, forQuery);
+  let recipientInfo = getRecipientNames();
+
+  function getRecipientNames() {
+    if (fromQuery !== undefined) {
+      let fromArray = fromQuery.split(',')
+      let recipientProfiles = []
+      if (fromArray.length !== 0) {
+        for (let i = 0; i < fromArray.length; ++i) {
+          let element = fromArray[i]
+          recipientProfiles.push(element)
+        }
+
+      } else {
+        recipientProfiles.push(fromQuery)
+
+      }
+      return recipientProfiles;
+
+    }
+
+  }
```

Comment:
Would it be more convenient to move the calls to `selectProfile` into the loop in this function? You could then return an array of the names instead an array of all the profile information.